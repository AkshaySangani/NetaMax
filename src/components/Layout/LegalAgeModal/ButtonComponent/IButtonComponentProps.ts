export interface IButtonComponentProps {
  /**
   * The label to display
   * @type {string}
   */
  label: string;

  /**
   * The onClick to modal
   * @type {void}
   */
  onClick: () => void;
}
