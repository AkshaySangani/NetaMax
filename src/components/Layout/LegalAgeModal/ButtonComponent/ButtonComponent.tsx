import React, { ReactElement } from "react";

import { Button } from "@chakra-ui/react";

import { IButtonComponentProps } from "./IButtonComponentProps";

/**
 * The Button component.
 * @param {IButtonComponentProps} props the component props.
 * @returns {ReactElement} the button component.
 */
export const ButtonComponent = ({ label, onClick }: IButtonComponentProps): ReactElement => {
  return (
    <Button
      color="#3870FF"
      bg="#FFFFFF"
      size="lg"
      onClick={onClick}
      px="42px"
      py="16px"
      fontWeight={700}
      borderRadius="16px"
    >
      {label}
    </Button>
  );
};
