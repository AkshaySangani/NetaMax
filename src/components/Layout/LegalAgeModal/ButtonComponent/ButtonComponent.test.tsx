import { render, fireEvent } from "@testing-library/react";
import { ButtonComponent } from "./ButtonComponent";

jest.mock("next/router", () => require("next-router-mock"));

describe("Render ButtonComponent component", () => {
  it("ButtonComponent should be render correctly", () => {
    const { container } = render(<ButtonComponent label="test" onClick={() => null} />);
    expect(container).toMatchSnapshot();
  });

  it("ButtonComponent should on click, when click in button", () => {
    const onClickMock = jest.fn();
    const { getByText } = render(<ButtonComponent label="test" onClick={onClickMock} />);
    fireEvent.click(getByText("test"));
    expect(onClickMock).toBeCalledTimes(1);
  });
});
