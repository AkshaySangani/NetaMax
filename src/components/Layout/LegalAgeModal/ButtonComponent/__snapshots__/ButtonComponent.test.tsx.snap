// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render ButtonComponent component ButtonComponent should be render correctly 1`] = `
<div>
  <button
    class="chakra-button css-1pgmkb6"
    type="button"
  >
    test
  </button>
</div>
`;
